instrumentation/vertx-reactive/src/main/java/io/opentelemetry/auto/instrumentation/vertx/reactive/AsyncResultConsumerWrapper.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.opentelemetry.auto.instrumentation.vertx.reactive;
+
+import static io.opentelemetry.auto.instrumentation.vertx.reactive.VertxDecorator.TRACER;
+
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.trace.Span;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import java.util.function.Consumer;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class AsyncResultConsumerWrapper implements Consumer<Handler<AsyncResult<?>>> {
+  private final Consumer<Handler<AsyncResult<?>>> delegate;
+  private final Span parentSpan;
```

Comment:
we should start propagating the `Context` instead of propagating (only) the parent `Span`, but this applies to lots of places, I'll open an issue for it

Reply:
opened #507